src/Microsoft.TestPlatform.CommunicationUtilities/Messages/ProtocolVersionAttribute.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.VisualStudio.TestPlatform.CommunicationUtilities.ObjectModel;
+
+[Conditional("DEBUG")]
+internal class ProtocolVersionAttribute : Attribute
```

Comment:
This is a nice addition, but I feel it needs some comments about its purpose and intended use so that it gets broader adoption.

Reply:
Yes I agree, the idea is not fully flashed out, and I asked other people for opinions and got none. So I kept what I have so far to be able to pilot this in VS asap.